### PR TITLE
6071 user account deletion ui

### LINKED ISSUE #6071: User acct deletion: modal overlay to confirm/cancel user deletion.
**Parent ticket:**
https://github.com/mdn/sprints/issues/1922

**Description:**
Overlay appears on MDN profile page when a logged-in MDN user clicks on 'Delete Account' link. 

**Acceptance Criteria:**
- [x] Modal needs to work on all devices. (for mobile - will go to a separate screen)
- [x] Clicking on 'Cancel' closes modal
- [x] Clicking on 'Yes, Delete my Account' executes deletion and closes modal. 

**Design:**
https://mozilla.invisionapp.com/share/Y8SGMKQBCW3#/screens/370705983_Account_-_Profile_-_Delete_Account_-no_Contrib-

### PR DESCRIPTION
**Pull request by schalkneethling**
fix #6071

@peterbe I have tested all of the flows using the modal as well as the dedicated page. The scenarios tested are:

1. No contributions, just delete the account (modal and page)
2. Keep contributions assigned to me and delete the account (modal and page)
3. Sign in again with the above account. This time assign attribution to Anonymous user and delete the account (modal and page)
4. Sign in, make some contributions, delete account donating contributions to Anonymous (modal and page)

r?

### CODE REVIEW

File: kuma/static/styles/components/compat-tables/_vars.scss [hunk 1]
```diff
@@ -43,12 +43,12 @@ $bc-mobile-browser-width: 220px;
...
+		size: 13,
 		feature-name-width: 200px,
 		platform-border: (8)
 	),
 	js: (
-		size: 13,
+		size: 14,
 		feature-name-width: 200px,
 		platform-border: (8, 15)
 	),
```
Comment (schalkneethling) [inline kuma/static/styles/components/compat-tables/_vars.scss]: I have no idea where this came from 🤔

Comment (schalkneethling) [reply kuma/static/styles/components/compat-tables/_vars.scss]: Gone now

**Comment by peterbe** [comment]
I think it's a bit weird that we have both now. 
On the Edit Profile page, if you right click and "Open in a new tab" you get to http://wiki.localhost.org:8000/en-US/profiles/peterbe/delete which still works. 

* Change that button looking `<a href="http://wiki.localhost.org:8000/en-US/profiles/peterbe/delete">` tag to an actual button. 
* http://wiki.localhost.org:8000/en-US/profiles/peterbe/delete should not work. Basically, the Django view I wrote should be git deleted.

**Comment by peterbe** [comment]
"Delete my Account" is a weird phrase. It should be "Delete my account". 

Screen Shot 2020-02-04 at 1 30 05 PM

File: kuma/static/js/components/account-management/delete-user-confirmation-button.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        return;
+    }
+
+    /* once a user select one of the attribution choices,
+        enable the submit button */
+    deleteUserForm.addEventListener('change', function(event) {
+        if (event.target.checked) {
+            deleteUserConfirm.classList.remove('disabled');
+            deleteUserConfirm.classList.add('negative');
+            deleteUserConfirm.removeAttribute('disabled');
```
Comment (peterbe) [inline kuma/static/js/components/account-management/delete-user-confirmation-button.js]: The button, when it's disabled, doesn't look like it's disabled. Not sure I care though :)

**Comment by atopal** [comment]
@schalkneethling can you also please add the username to "Keep my attribution for my page changes (username)"

**Comment by schalkneethling** [comment]
> @schalkneethling can you also please add the username to "Keep my attribution for my page changes (username)"

The reason it is not there is that we do not know the username in the context.
https://github.com/mdn/kuma/pull/6378/files#diff-43b938f2de0dfde7bfcdcd718f164865R211

**Comment by schalkneethling** [comment]
> I think it's a bit weird that we have both now.
> On the Edit Profile page, if you right click and "Open in a new tab" you get to http://wiki.localhost.org:8000/en-US/profiles/peterbe/delete which still works.
> 

Well, on mobile it will go to this page(and if you open it in a new tab).

> * Change that button looking `<a href="http://wiki.localhost.org:8000/en-US/profiles/peterbe/delete">` tag to an actual button.

Technically this link does take you somewhere, it is just that JavaScript hijacks it if enabled, and you are not on mobile. I reckon I link still makes sense.

> * http://wiki.localhost.org:8000/en-US/profiles/peterbe/delete should not work. Basically, the Django view I wrote should be git deleted.

As mentioned above, we do use it on mobile, if you choose to open in a new tab, and also should the JS fail for whatever reason(or is disabled)

**Comment by schalkneethling** [comment]
> @schalkneethling can you also please add the username to "Keep my attribution for my page changes (username)"

Is this a high priority requirement @atopal? If so, we can make it work. Let me know, thanks!

**Comment by schalkneethling** [comment]
> "Delete my Account" is a weird phrase. It should be "Delete my account".

I agree it is a bit odd. I followed what was in the UX designs.

File: kuma/static/js/components/account-management/delete-user-confirmation-button.js (hunk 2 above)
Comment (schalkneethling) [reply kuma/static/js/components/account-management/delete-user-confirmation-button.js]: Really? Looks disabled here https://github.com/mdn/kuma/pull/6378#issuecomment-582051746 and there is also no little hand when you hover over it. In fact, it does not even have a hover state or respond to pointer events.

**Comment by schalkneethling** [comment]
Thank you for the review @peterbe - I will wait for answers and comments on some of the points here. If all is well, I will go ahead and merge.

**Comment by peterbe** [comment]
> > @schalkneethling can you also please add the username to "Keep my attribution for my page changes (username)"
> 
> The reason it is not there is that we do not know the username in the context.
> https://github.com/mdn/kuma/pull/6378/files#diff-43b938f2de0dfde7bfcdcd718f164865R211

I see! The text is not typed in a template but deep inside a form class. 
I think it's fine to deviate from the designs and use good judgement.

**Comment by peterbe** [comment]
> > "Delete my Account" is a weird phrase. It should be "Delete my account".
> 
> I agree it is a bit odd. I followed what was in the UX designs.

Neither you or me are native English speakers or writers. But "Delete my Account" looks German to me with the noun being capitalized. I think it's different if it's verbs peppered with stop words. Like "Publish and Edit".

It's 0.1% important for the sake of the PR, but it's progressive to take the time to upgrade our writing skills so there's 5% in debating it. If you don't have a burning passion for this, just ignore it.

**Comment by atopal** [comment]
It's called title case, and it's used for buttons in UIs. As a UX professional that's what Nick was probably going for, but he didn't capitalize the personal pronoun. It should read "Yes, Delete My Account". That said, title case is hard to read beyond 3 or so words, so this is borderline. Either way, please don't use "Yes, Delete my account"

@schalkneethling mentioning the username was a deliberate usability decision between me and Nick. If there is an issue with something like that, please bring it up in the issue or the daily stand up, but please don't just drop it like that. In this case, I'd like to understand the effort necessary to get that context, and if that's too high, we can drop it.

Btw, a textbook case for why you want to implement back and front-end in tandem ;)

**Comment by schalkneethling** [comment]
@atopal I have created the following two issues as follow on items from this work:

- https://github.com/mdn/kuma/issues/6418
- https://github.com/mdn/kuma/issues/6419

**Pull request merged**